Fix resource leaks in Input Actions Editor window.

AlexTyrer opened the pull request:
### Description

This PR addresses resource leaks in the Input Actions Editor window.

These resource leaks could lead to unexpected exceptions being thrown.

### Changes made

InputActionsEditorSettingsProvider is now handled by a single static instance to work around UIToolkit callback order issues identified.

Fix VisualElement handling on the StateContainer that manages the tracked serialized object representing the Input Actions.


### Checklist

Before review:

- [x] Changelog entry added.
    - Explains the change in `Changed`, `Fixed`, `Added` sections.
    - For API change contains an example snippet and/or migration example.
    - FogBugz ticket attached, example `([case %number%](https://issuetracker.unity3d.com/issues/...))`.
    - FogBugz is marked as "Resolved" with *next* release version correctly set.
- [x] Tests added/changed, if applicable.
    - Functional tests `Area_CanDoX`, `Area_CanDoX_EvenIfYIsTheCase`, `Area_WhenIDoX_AndYHappens_ThisIsTheResult`.
    - Performance tests.
    - Integration tests.
- [x] Docs for new/changed API's.
    - Xmldoc cross references are set correctly.
    - Added explanation how the API works.
    - Usage code examples added.
    - The manual is updated, if needed.

- [x] Commit message for squash-merge is prefixed with one of the list:
    - FIX: Fixed Input Actions Editor window resource leak that could result in unexpected exceptions [ISXB-865](https://jira.unity3d.com/browse/ISXB-865)

AlexTyrer committed:
[Input] Fix for UI invocations during test runs causing exceptions (event handler leaks)

AlexTyrer committed:
[Input] Fix Input Action Editor resource leak that could result in exceptions being thrown

AlexTyrer committed:
[Input] Update changelog to show fix for ISXB-865

ekcoh commented on Packages/com.unity.inputsystem/CHANGELOG.md:
```diff
@@ -17,6 +17,7 @@ however, it has to be formatted properly to pass verification tests.
 - Fixed an issue where `System.ObjectDisposedException` would be thrown when deleting the last ActionMap item in the Input Actions Asset editor.
 - Fixed DualSense Edge's vibration and light bar not working on Windows
 - Fixed Project-wide Actions asset failing to reload properly after deleting project's Library folder.
+- Fixed Input Actions Editor window resource leak that could result in unexpected exceptions [ISXB-865](https://jira.unity3d.com/browse/ISXB-865)
```
This should not use internal URL. If internal only provide issue reference number or skip.

AlexTyrer replied:
Of course - I've removed the internal link in 5e872116617b56d20a930efac85404e519874a53

ekcoh commented on Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/StateContainer.cs:
```diff
@@ -51,9 +43,17 @@ public void Dispatch(Command command)
-        public void Initialize()
+        public void Initialize(VisualElement rootVisualElement)
         {
+            m_RootVisualElement = rootVisualElement;
```
Maybe add a comment here explaining why we do this little dance?
E.g. "We need to use a root element for TrackSerializedObjectValue that is destroyed with the view. Using a root element from the settings window would not enable the tracking callback to be destroyed or garbage collected."

AlexTyrer replied:
Good idea - have added the comment in 5e872116617b56d20a930efac85404e519874a53

ekcoh reviewed (changes requested):
Two small comments to address, otherwise LGTM. Will approve once updated if QA is positive.

AlexTyrer committed:
[Input] Address PR feedback

o Removed internal JIRA link (just leaving number for reference)
o Added a comment explaining the reason for the root visual element handling

ekcoh commented:
I think this can land after QA assessment is available with approval.

Pauliusd01 reviewed (approved):
Since I wasn't able to repro the original issues in Unity 6 I only checked basic ProjectWideActions and custom actions window functionality. Did not notice any new issues from these changes. LGTM

ekcoh committed:
Merge branch 'develop' into ISX-1962-ISXB-865-fix-input-resource-leaks
